shared/src/simulation/behaviors/request.ts

```diff
@@ -167,43 +146,62 @@ export const requestBehavior: SimulationBehavior<RequestBehaviorState> = {
                             Object.keys(promise.resource.vehicleCounts).length >
                             0
                     );
+
+                behaviorState.requestedResources = {};
                 break;
             }
             case 'sendRequestEvent': {
-                if (!isWaitingForAnswer(behaviorState)) {
-                    const resourcesToRequest = getResourcesToRequest(
-                        draftState,
-                        simulatedRegion,
-                        behaviorState
-                    );
-                    const resource = VehicleResource.create(
-                        resourcesToRequest as ResourceDescription
-                    );
-                    if (!isEmptyResource(resource)) {
-                        // create a request to wait for an answer
-                        behaviorState.answerKey = `${simulatedRegion.id}-request-${behaviorState.requestTargetVersion}`;
-                        const activityId = nextUUID(draftState);
-                        addActivity(
-                            simulatedRegion,
-                            CreateRequestActivityState.create(
-                                activityId,
-                                behaviorState.requestTarget,
-                                resource,
-                                behaviorState.answerKey
-                            )
-                        );
-                    }
-                }
+                const resourcesToRequest = getResourcesToRequest(
+                    draftState,
+                    behaviorState
+                );
+                const resource = VehicleResource.create(
+                    resourcesToRequest as ResourceDescription
+                );
+                addActivity(
+                    simulatedRegion,
+                    CreateRequestActivityState.create(
+                        nextUUID(draftState),
+                        behaviorState.requestTarget,
+                        resource,
+                        `${simulatedRegion.id}-request`
```

Comment:
You should use a function to generate this key to enforce consistency.

Reply:
Done in 879cb599205c398487892dd7c8f580d0e1f50886